Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -69,11 +68,24 @@ Status& Status::WithPlatformError(std::unique_ptr<PlatformError> error) {
   return *this;
 }
 
+bool Status::IsMoved() const {
+  return IsStatePtrMoved(state_);
+}
+
+void Status::SetMoved() {
+  if (IsMoved()) {
```

Comment:
I don't think this check buys us anything. Constructing a moved-from pointer is very cheap, probably cheaper than having conditional logic. From the correctness perspective, I can't see any issue with setting a moved-from pointer to the same special value again.